packages/contracts/src/utils/exchange_wrapper.ts

```diff
@@ -35,7 +35,11 @@ export class ExchangeWrapper {
         );
         const tx = await this._zeroEx.awaitTransactionMinedAsync(txHash);
         tx.logs = _.filter(tx.logs, log => log.address === this._exchange.address);
-        _.each(tx.logs, log => wrapLogBigNumbers(log));
+        tx.logs = _.map(tx.logs, log => {
+            const decodedLog = this._logDecoder.tryToDecodeLogOrNoop(log);
```

Comment:
So this is not correct handling of the return value of the method, **unless** you are absolutely certain that given your log, it will always return the decoded log. Are you?